interfaces.d.ts

```diff
@@ -31,6 +31,47 @@ export interface PolymerInit {
   listeners?: {[key: string]: string};
 }
 
+/**
+ * The object passed to ".*" wildcard obsevers. A record of a change made to an
+ * object.
+ * @template B The object matching the non-wildcard portion of the path.
+ * @template V Additional types that could be set at the path.
+ */
+export interface PolymerDeepPropertyChange<B, V> {
+  /** Path to the property that changed. */
+  path: string;
+  /** The object matching the non-wildcard portion of the path. */
+  base: B;
+  /** New value of the path that changed. */
+  value: B|V;
```

Comment:
Why B|V and not just V?

Reply:
In the change record, `base` will always be the base of the wildcard path (e.g.if you observe `foo.bar.*` then `base` will always be `foo.bar`), while `value` could be the base, or any other path nested below it (e.g. `foo.bar`, then `foo.bar.baz`). So the idea is you specify the type of the base path, and also a union of whatever other types you know you could encounter deeply within it:

`fooChanged(changeRecord: PolymerDeepPropertyChanged<MyElement['foo'], string|undefined>)`

Reply:
To more directly answer your question, it's `B|V` because you always get one notification with the base in the `value` field, and again every time the base is re-set.

Reply:
I see. That's really hard to typecheck!